### PR TITLE
Improve serialisation in analysis

### PR DESCRIPTION
**Pull request by QRemy**
This PR implements a part of the features proposed in #3788 related to datasets serialisation  :
- add datasets_file and models_file to general config
- add read_datasets write_datasets methods on analysis
- add extend option to set_models so extending the exiting models is not hard-coded
- change analysis.models to a property returning datasets.models so they are always the same and add a models setter on analysis that call set_models(models, extend=False)

### CODE REVIEW

**Review by registerrier — COMMENTED** [review]
Thanks @QRemy !

I have left a number of comments inline.
My main comment is whether methods defined on `Analysis` should take arguments defined in the `config` file. I would think not.

Also, where is the best place to put the datasets file name? One could argue that it would fit better into the Datasets configuration. Most likely not for reading, but for dataset creation certainly.

File: gammapy/analysis/core.py [hunk 1]
```diff
@@ -155,45 +164,125 @@ def get_datasets(self):
         else:  # 3d
             self._map_making()
 
-    def set_models(self, models):
-        """Set models on datasets.
 
+    def set_models(self, models, extend=True):
```
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/analysis/core.py:169]: same as previously automatic background model creation is done only if not defined in the datasets after updating the models

File: gammapy/analysis/core.py [hunk 2]
```diff
@@ -155,45 +164,125 @@ def get_datasets(self):
...
+            self.datasets = None
 
-        log.info(self.models)
 
-    def read_models(self, path):
-        """Read models from YAML file."""
-        path = make_path(path)
-        models = Models.read(path)
-        self.set_models(models)
+    def write_datasets(self, filename=None, filename_models=None):
```
[the remark this answers is not in the document]
Comment (QRemy) [reply gammapy/analysis/core.py:250]: ok removed filenames

**Comment by QRemy** [comment]
> Also, where is the best place to put the datasets file name? One could argue that it would fit better into the Datasets configuration. Most likely not for reading, but for dataset creation certainly.

The problem I see here is that the `DatasetsConfig` we have is more a `DatasetConfig` and I would prefer to work on the config structure in a separate PR.

**Review by registerrier — APPROVED** [review]
Thanks @QRemy. This looks good!

**Pull request merged**